File: LeetCode/Java Solutions/Largest 3-Same-Digit Number in String/largest.java
public class Solution {
    public String largestGoodInteger(String num) {
        String maxGood = "";

        for (int i = 0; i < num.length() - 2; i++) {
            // Check if the substring of length 3 has the same character
            if (num.charAt(i) == num.charAt(i + 1) && num.charAt(i) == num.charAt(i + 2)) {
                // Update maxGood if the current substring is greater
                maxGood = max(maxGood, num.substring(i, i + 3));
            }
        }

        return maxGood;
    }

    private String max(String a, String b) {
        return a.compareTo(b) > 0 ? a : b;
    }
}

File: LeetCode/Python Solutions/Largest 3-Same-Digit Number in String/largest.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        max_good = ""

        for i in range(len(num) - 2):
            # Check if the substring of length 3 has the same character
            if num[i] == num[i + 1] == num[i + 2]:
                # Update max_good if the current substring is greater
                max_good = max(max_good, num[i:i + 3])

        return max_good
